src/OrleansCodeGenerator/SerializerGenerator.cs

```diff
@@ -480,13 +477,52 @@ private static ExpressionSyntax GetObjectCreationExpressionSyntax(Type type)
             else
             {
                 // Create an unformatted object.
-                Expression<Func<object>> getUninitializedObject =
+#if DNXCORE50
+                var bindingFlags = SyntaxFactoryExtensions.GetBindingFlagsParenthesizedExpressionSyntax(
+                    SyntaxKind.BitwiseOrExpression,
+                    BindingFlags.Instance,
+                    BindingFlags.NonPublic,
+                    BindingFlags.Public);
+
+                var activatorCreateInstanceParamsCount = 5;
+                ArgumentSyntax[] createInstanceArguments = new ArgumentSyntax[activatorCreateInstanceParamsCount];
+                var nullLiteralExpressionArgument = SF.Argument(SF.LiteralExpression(SyntaxKind.NullLiteralExpression));
+                createInstanceArguments[0] = SF.Argument(SF.TypeOfExpression(typeInfo.GetTypeSyntax()));
+                createInstanceArguments[1] = SF.Argument(bindingFlags);
+                createInstanceArguments[2] = nullLiteralExpressionArgument;
+                var cons = typeInfo.GetConstructors(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance)
+                    .OrderBy(p => p.GetParameters().Count()).FirstOrDefault();
+                var consParameters = cons != null ? cons.GetParameters().Select(p => p.ParameterType).ToArray() : null;
+                if (consParameters != null && consParameters.Length > 0)
+                {
+                    var separatedSyntaxList = new SeparatedSyntaxList<ExpressionSyntax>();
+                    separatedSyntaxList = consParameters
+                        .Aggregate(separatedSyntaxList, (current, t) => current.Add(SF.DefaultExpression(t.GetTypeInfo().GetTypeSyntax())));
+
+                    createInstanceArguments[3] = SF.Argument(separatedSyntaxList.GetArrayCreationWithInitializerSyntax(
+                        SF.PredefinedType(SF.Token(SyntaxKind.ObjectKeyword))));
+                }
+                else
+                {
+                    createInstanceArguments[3] = nullLiteralExpressionArgument;
+                }
+
+                createInstanceArguments[4] = nullLiteralExpressionArgument;
+                Expression<Func<object>> getUninitializedObject = () => Activator.CreateInstance(default(Type));
```

Comment:
What kind of constructors do you mean? The case of nonpublic ones is covered with ```BindingFlags.NonPublic```.